### PR TITLE
metrics: Add db level metrics for server/metrics

### PR DESCRIPTION
**Pull request by ming-relax**
Added `db` label to some metrics. It is easier for developer to identity issues if we can see metrics in different db.

Only server metrics and distsql related metrics are added right now.

Users can choose to separate metrics by `db` or they can just use the old global metrics.

### CODE REVIEW

**Comment by ngaut** [comment]
Cool. Maybe there are some cross database queries.
But we can focus on a single database right now.

**Comment by ming-relax** [comment]
@shenli can you remove the `wip` tag for this pr?

File: executor/distsql.go [hunk 1]
```diff
@@ -668,7 +668,7 @@ func (e *XSelectIndexExec) doIndexRequest() (distsql.SelectResult, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return distsql.Select(e.ctx.GetClient(), e.ctx.GoCtx(), selIdxReq, keyRanges, e.scanConcurrency, !e.outOfOrder, getIsolationLevel(sv), e.priority)
+	return distsql.Select(e.ctx.GetSessionVars().CurrentDB, e.ctx.GetClient(), e.ctx.GoCtx(), selIdxReq, keyRanges, e.scanConcurrency, !e.outOfOrder, getIsolationLevel(sv), e.priority)
```
Comment (shenli) [inline executor/distsql.go]: CurrentDB is the default database, but user could choose another database in their SQL. For example, `use test; ...; select * from test1.t;`.

**Comment by ming-relax** [comment]
@shenli seems not easy to extract the db name from the current sql, we can add the metrics to `server` module first.

**Comment by ming-relax** [comment]
@shenli any further comments on this pr ?

**Comment by shenli** [comment]
@ming-relax We will take a look.

**Comment by coocood** [comment]
LGTM

**Comment by coocood** [comment]
@shenli  PTAL

File: server/conn.go [hunk 2]
```diff
@@ -418,7 +418,7 @@ func (cc *clientConn) Run() {
 			} else if terror.ErrCritical.Equal(err) {
 				log.Errorf("[%d] critical error, stop the server listener %s",
 					cc.connectionID, errors.ErrorStack(err))
-				criticalErrorCounter.Add(1)
+				criticalErrorCounter.WithLabelValues("db").Add(1)
```
Comment (zimulala) [inline server/conn.go]: Why we don't use `cc.dbname` here?

**Comment by zhexuany** [comment]
@ming-relax friendly ping. Please address comments.

**Comment by coocood** [comment]
@ming-relax 
It has been a long time since the last update, so we just close this PR for now, we can reopen it later if you update this PR.

**Pull request closed without merging**